lib/compiler-singlepass/src/emitter_x64.rs

```diff
@@ -909,11 +909,12 @@ macro_rules! sse_round_fn {
         |emitter: &mut AssemblerX64, precision: Precision, src1: XMM, src2: XMMOrMemory, dst: XMM| {
             match src2 {
                 XMMOrMemory::XMM(x) => {
-                    assert_eq!(src1, x);
-                    move_src_to_dst(emitter, precision, src1, dst);
+                    move_src_to_dst(emitter, precision, x, dst);
```

Comment:
No like this.
You need to create an `if src1 == x` with that new code, and have the old code in the `else` branch.